mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,530 @@
+    origin : array-like, shape (3,)
+        Origin of internal and external multipolar moment space in head coords
+        and in millimeters
+    int_order : int
+        Order of internal component of spherical expansion
+    ext_order : int
+        Order of external component of spherical expansion
+
+    Returns
+    -------
+    raw_sss : instance of mne.io.Raw
+        The raw data with Maxwell filtering applied
+
+    Notes
+    -----
+    .. versionadded:: 0.10
+
+    Equation numbers refer to Taulu and Kajola, 2005.
+
+    There are an absurd number of different possible notations for spherical
+    coordinates, which confounds the notation for spherical harmonics.  Here,
+    we purposefully stay away from shorthand notation in both and use explicit
+    terms (like 'azimuth' and 'polar') to avoid confusion.
+
+    This code was adapted and relicensed (with BSD form) with permission from
+    Jussi Nurminen.
+
+    References
+    ----------
+    .. [1] Taulu and Kajola, 2005. "Presentation of electromagnetic
+           multichannel data: The signal space separation method".
+           http://lib.tkk.fi/Diss/2008/isbn9789512295654/article2.pdf
+    """
+
+    # TODO: Exclude 'bads' in multipolar moment calc, add back during
+    # reconstruction
+    if len(raw.info['bads']) > 0:
+        raise RuntimeError('Maxwell filter does not yet handle bad channels.')
+
+    # TODO: Improve logging process to better match Elekta's version
+    # Read coil definitions from file
+    all_coils, meg_info = _make_coils(raw.info, accurate=True,
+                                      elekta_defs=True)
+
+    # Create coil list and pick MEG channels
+    picks = [raw.info['ch_names'].index(coil['chname'])
+             for coil in all_coils]
+    coils = [all_coils[ci] for ci in picks]
+    raw.pick_channels(coil['chname'] for coil in coils)
+
+    data, times = raw[picks, :]
+
+    # Magnetometers (with coil_class == 1.0) must be scaled by 100 to improve
+    # numerical stability as they have different scales than gradiometers
+    coil_scale = np.ones(len(picks))
+    coil_scale[np.array([coil['coil_class'] == 1.0 for coil in coils])] = 100.
+
+    # Compute multipolar moment bases
+    origin = np.array(origin) / 1000.  # Convert scale from mm to m
+    S_in, S_out = _sss_basis(origin, coils, int_order, ext_order)
+    S_tot = np.c_[S_in, S_out]
+
+    # Pseudo-inverse of total multipolar moment basis set (Part of Eq. 37)
+    pS_tot = pinv(S_tot, cond=1e-15)
+    # Compute multipolar moments of (magnetometer scaled) data (Eq. 37)
+    mm = np.dot(pS_tot, data * coil_scale[:, np.newaxis])
+    # Reconstruct data from internal space (Eq. 38)
+    recon = np.dot(S_in, mm[:S_in.shape[1], :])
+
+    # Return reconstructed raw file object
+    raw_sss = _update_info(raw.copy(), origin, int_order, ext_order,
+                           data.shape[0], mm.shape[0])
+    raw_sss._data[:, :] = recon / coil_scale[:, np.newaxis]
```

Comment:
Oh wait -- on second thought, you don't want to do `raw.pick_channels` above. Think of the example of MEEG data, what you output now will only have MEG data (will be stripped of EEG data), which is wrong. You'll also get rid of EOG, ECG, triggers, ... Instead you should just process and overwrite the MEG entries in `_data` using the `picks` attribute.